### PR TITLE
Add unit test for cdef

### PR DESCRIPTION
**Pull request by Cidana-Developers**
Add test for cdef_find_dir, cdef_filter_block, copy_rect8_8bit_to_16bit,
compute_cdef_dist and search_one_dual functions.

### CODE REVIEW

**Comment by Cidana-Developers** [comment]
# Pr for Cdef test
  updated on 2019-07-15
## warnings in Local build
1. mac:
One warning, `ld: warning: -pie being ignored. It is only used when linking a main executable`, can be ignored.
2. linux:
One warning, can be ignored:
`make[4]: warning: jobserver unavailable: using -j1.  Add '+' to parent make rule.`
3. windows:  
Several warnings:
```
x86inc.obj : warning LNK4221: This object file does not define any previously undefined public symbols, so it will not
be used by any link operation that consumes this library [F:\work\SVT-AV1\Build\windows\Source\Lib\Common\ASM_SSE2\COMM
ON_ASM_SSE2.vcxproj]
x86_abi_support.obj : warning LNK4221: This object file does not define any previously undefined public symbols, so it
will not be used by any link operation that consumes this library [F:\work\SVT-AV1\Build\windows\Source\Lib\Common\ASM_
SSE2\COMMON_ASM_SSE2.vcxproj]
x64inc.obj : warning LNK4221: This object file does not define any previously undefined public symbols, so it will not
be used by any link operation that consumes this library [F:\work\SVT-AV1\Build\windows\Source\Lib\Common\ASM_SSE2\COMM
ON_ASM_SSE2.vcxproj]
x64Macro.obj : warning LNK4221: This object file does not define any previously undefined public symbols, so it will no
t be used by any link operation that consumes this library [F:\work\SVT-AV1\Build\windows\Source\Lib\Common\ASM_SSE2\CO
MMON_ASM_SSE2.vcxproj]
```

## warnings on remote ci servers
1. jenkins:
a. clang-format
b. 
2. travis
  a. mac: 
  One warning, can be ignored.
  ```
  ld: warning: -pie being ignored. It is only used when linking a main executable  
  ```
  b. linux: 
  No warning.

3. appvedor
  a. no warnings

## dry run output
1. cdef block run long time;
```
[==========] Running 196 tests from 3 test cases.
[----------] Global test environment set-up.
[----------] 3 tests from CdefToolTest
[ RUN      ] CdefToolTest.CopyRectMatchTest
[       OK ] CdefToolTest.CopyRectMatchTest (3 ms)
[ RUN      ] CdefToolTest.ComputeCdefDistMatchTest
[       OK ] CdefToolTest.ComputeCdefDistMatchTest (131 ms)
[ RUN      ] CdefToolTest.SearchOneDualMatchTest
[       OK ] CdefToolTest.SearchOneDualMatchTest (877 ms)
[----------] 3 tests from CdefToolTest (1011 ms total)

[----------] 192 tests from Cdef/CDEFBlockTest
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/0
[       OK ] Cdef/CDEFBlockTest.MatchTest/0 (46362 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/1
[       OK ] Cdef/CDEFBlockTest.MatchTest/1 (69448 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/2
[       OK ] Cdef/CDEFBlockTest.MatchTest/2 (83670 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/3
[       OK ] Cdef/CDEFBlockTest.MatchTest/3 (50 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/4
[       OK ] Cdef/CDEFBlockTest.MatchTest/4 (79 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/5
[       OK ] Cdef/CDEFBlockTest.MatchTest/5 (79 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/6
[       OK ] Cdef/CDEFBlockTest.MatchTest/6 (49 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/7
[       OK ] Cdef/CDEFBlockTest.MatchTest/7 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/8
[       OK ] Cdef/CDEFBlockTest.MatchTest/8 (79 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/9
[       OK ] Cdef/CDEFBlockTest.MatchTest/9 (49 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/10
[       OK ] Cdef/CDEFBlockTest.MatchTest/10 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/11
[       OK ] Cdef/CDEFBlockTest.MatchTest/11 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/12
[       OK ] Cdef/CDEFBlockTest.MatchTest/12 (49 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/13
[       OK ] Cdef/CDEFBlockTest.MatchTest/13 (79 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/14
[       OK ] Cdef/CDEFBlockTest.MatchTest/14 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/15
[       OK ] Cdef/CDEFBlockTest.MatchTest/15 (49 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/16
[       OK ] Cdef/CDEFBlockTest.MatchTest/16 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/17
[       OK ] Cdef/CDEFBlockTest.MatchTest/17 (79 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/18
[       OK ] Cdef/CDEFBlockTest.MatchTest/18 (49 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/19
[       OK ] Cdef/CDEFBlockTest.MatchTest/19 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/20
[       OK ] Cdef/CDEFBlockTest.MatchTest/20 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/21
[       OK ] Cdef/CDEFBlockTest.MatchTest/21 (49 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/22
[       OK ] Cdef/CDEFBlockTest.MatchTest/22 (81 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/23
[       OK ] Cdef/CDEFBlockTest.MatchTest/23 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/24
[       OK ] Cdef/CDEFBlockTest.MatchTest/24 (48 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/25
[       OK ] Cdef/CDEFBlockTest.MatchTest/25 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/26
[       OK ] Cdef/CDEFBlockTest.MatchTest/26 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/27
[       OK ] Cdef/CDEFBlockTest.MatchTest/27 (48 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/28
[       OK ] Cdef/CDEFBlockTest.MatchTest/28 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/29
[       OK ] Cdef/CDEFBlockTest.MatchTest/29 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/30
[       OK ] Cdef/CDEFBlockTest.MatchTest/30 (48 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/31
[       OK ] Cdef/CDEFBlockTest.MatchTest/31 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/32
[       OK ] Cdef/CDEFBlockTest.MatchTest/32 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/33
[       OK ] Cdef/CDEFBlockTest.MatchTest/33 (49 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/34
[       OK ] Cdef/CDEFBlockTest.MatchTest/34 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/35
[       OK ] Cdef/CDEFBlockTest.MatchTest/35 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/36
[       OK ] Cdef/CDEFBlockTest.MatchTest/36 (49 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/37
[       OK ] Cdef/CDEFBlockTest.MatchTest/37 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/38
[       OK ] Cdef/CDEFBlockTest.MatchTest/38 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/39
[       OK ] Cdef/CDEFBlockTest.MatchTest/39 (48 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/40
[       OK ] Cdef/CDEFBlockTest.MatchTest/40 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/41
[       OK ] Cdef/CDEFBlockTest.MatchTest/41 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/42
[       OK ] Cdef/CDEFBlockTest.MatchTest/42 (49 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/43
[       OK ] Cdef/CDEFBlockTest.MatchTest/43 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/44
[       OK ] Cdef/CDEFBlockTest.MatchTest/44 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/45
[       OK ] Cdef/CDEFBlockTest.MatchTest/45 (49 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/46
[       OK ] Cdef/CDEFBlockTest.MatchTest/46 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/47
[       OK ] Cdef/CDEFBlockTest.MatchTest/47 (80 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/48
[       OK ] Cdef/CDEFBlockTest.MatchTest/48 (82096 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/49
[       OK ] Cdef/CDEFBlockTest.MatchTest/49 (119638 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/50
[       OK ] Cdef/CDEFBlockTest.MatchTest/50 (142742 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/51
[       OK ] Cdef/CDEFBlockTest.MatchTest/51 (85 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/52
[       OK ] Cdef/CDEFBlockTest.MatchTest/52 (134 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/53
[       OK ] Cdef/CDEFBlockTest.MatchTest/53 (134 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/54
[       OK ] Cdef/CDEFBlockTest.MatchTest/54 (89 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/55
[       OK ] Cdef/CDEFBlockTest.MatchTest/55 (131 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/56
[       OK ] Cdef/CDEFBlockTest.MatchTest/56 (131 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/57
[       OK ] Cdef/CDEFBlockTest.MatchTest/57 (84 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/58
[       OK ] Cdef/CDEFBlockTest.MatchTest/58 (131 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/59
[       OK ] Cdef/CDEFBlockTest.MatchTest/59 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/60
[       OK ] Cdef/CDEFBlockTest.MatchTest/60 (83 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/61
[       OK ] Cdef/CDEFBlockTest.MatchTest/61 (131 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/62
[       OK ] Cdef/CDEFBlockTest.MatchTest/62 (131 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/63
[       OK ] Cdef/CDEFBlockTest.MatchTest/63 (83 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/64
[       OK ] Cdef/CDEFBlockTest.MatchTest/64 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/65
[       OK ] Cdef/CDEFBlockTest.MatchTest/65 (132 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/66
[       OK ] Cdef/CDEFBlockTest.MatchTest/66 (88 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/67
[       OK ] Cdef/CDEFBlockTest.MatchTest/67 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/68
[       OK ] Cdef/CDEFBlockTest.MatchTest/68 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/69
[       OK ] Cdef/CDEFBlockTest.MatchTest/69 (84 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/70
[       OK ] Cdef/CDEFBlockTest.MatchTest/70 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/71
[       OK ] Cdef/CDEFBlockTest.MatchTest/71 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/72
[       OK ] Cdef/CDEFBlockTest.MatchTest/72 (83 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/73
[       OK ] Cdef/CDEFBlockTest.MatchTest/73 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/74
[       OK ] Cdef/CDEFBlockTest.MatchTest/74 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/75
[       OK ] Cdef/CDEFBlockTest.MatchTest/75 (83 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/76
[       OK ] Cdef/CDEFBlockTest.MatchTest/76 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/77
[       OK ] Cdef/CDEFBlockTest.MatchTest/77 (132 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/78
[       OK ] Cdef/CDEFBlockTest.MatchTest/78 (85 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/79
[       OK ] Cdef/CDEFBlockTest.MatchTest/79 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/80
[       OK ] Cdef/CDEFBlockTest.MatchTest/80 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/81
[       OK ] Cdef/CDEFBlockTest.MatchTest/81 (84 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/82
[       OK ] Cdef/CDEFBlockTest.MatchTest/82 (131 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/83
[       OK ] Cdef/CDEFBlockTest.MatchTest/83 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/84
[       OK ] Cdef/CDEFBlockTest.MatchTest/84 (83 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/85
[       OK ] Cdef/CDEFBlockTest.MatchTest/85 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/86
[       OK ] Cdef/CDEFBlockTest.MatchTest/86 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/87
[       OK ] Cdef/CDEFBlockTest.MatchTest/87 (84 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/88
[       OK ] Cdef/CDEFBlockTest.MatchTest/88 (131 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/89
[       OK ] Cdef/CDEFBlockTest.MatchTest/89 (131 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/90
[       OK ] Cdef/CDEFBlockTest.MatchTest/90 (84 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/91
[       OK ] Cdef/CDEFBlockTest.MatchTest/91 (131 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/92
[       OK ] Cdef/CDEFBlockTest.MatchTest/92 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/93
[       OK ] Cdef/CDEFBlockTest.MatchTest/93 (84 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/94
[       OK ] Cdef/CDEFBlockTest.MatchTest/94 (135 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/95
[       OK ] Cdef/CDEFBlockTest.MatchTest/95 (131 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/96
[       OK ] Cdef/CDEFBlockTest.MatchTest/96 (84302 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/97
[       OK ] Cdef/CDEFBlockTest.MatchTest/97 (120692 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/98
[       OK ] Cdef/CDEFBlockTest.MatchTest/98 (145114 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/99
[       OK ] Cdef/CDEFBlockTest.MatchTest/99 (91 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/100
[       OK ] Cdef/CDEFBlockTest.MatchTest/100 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/101
[       OK ] Cdef/CDEFBlockTest.MatchTest/101 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/102
[       OK ] Cdef/CDEFBlockTest.MatchTest/102 (82 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/103
[       OK ] Cdef/CDEFBlockTest.MatchTest/103 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/104
[       OK ] Cdef/CDEFBlockTest.MatchTest/104 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/105
[       OK ] Cdef/CDEFBlockTest.MatchTest/105 (83 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/106
[       OK ] Cdef/CDEFBlockTest.MatchTest/106 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/107
[       OK ] Cdef/CDEFBlockTest.MatchTest/107 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/108
[       OK ] Cdef/CDEFBlockTest.MatchTest/108 (82 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/109
[       OK ] Cdef/CDEFBlockTest.MatchTest/109 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/110
[       OK ] Cdef/CDEFBlockTest.MatchTest/110 (128 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/111
[       OK ] Cdef/CDEFBlockTest.MatchTest/111 (83 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/112
[       OK ] Cdef/CDEFBlockTest.MatchTest/112 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/113
[       OK ] Cdef/CDEFBlockTest.MatchTest/113 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/114
[       OK ] Cdef/CDEFBlockTest.MatchTest/114 (82 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/115
[       OK ] Cdef/CDEFBlockTest.MatchTest/115 (133 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/116
[       OK ] Cdef/CDEFBlockTest.MatchTest/116 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/117
[       OK ] Cdef/CDEFBlockTest.MatchTest/117 (83 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/118
[       OK ] Cdef/CDEFBlockTest.MatchTest/118 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/119
[       OK ] Cdef/CDEFBlockTest.MatchTest/119 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/120
[       OK ] Cdef/CDEFBlockTest.MatchTest/120 (82 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/121
[       OK ] Cdef/CDEFBlockTest.MatchTest/121 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/122
[       OK ] Cdef/CDEFBlockTest.MatchTest/122 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/123
[       OK ] Cdef/CDEFBlockTest.MatchTest/123 (82 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/124
[       OK ] Cdef/CDEFBlockTest.MatchTest/124 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/125
[       OK ] Cdef/CDEFBlockTest.MatchTest/125 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/126
[       OK ] Cdef/CDEFBlockTest.MatchTest/126 (82 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/127
[       OK ] Cdef/CDEFBlockTest.MatchTest/127 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/128
[       OK ] Cdef/CDEFBlockTest.MatchTest/128 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/129
[       OK ] Cdef/CDEFBlockTest.MatchTest/129 (83 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/130
[       OK ] Cdef/CDEFBlockTest.MatchTest/130 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/131
[       OK ] Cdef/CDEFBlockTest.MatchTest/131 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/132
[       OK ] Cdef/CDEFBlockTest.MatchTest/132 (82 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/133
[       OK ] Cdef/CDEFBlockTest.MatchTest/133 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/134
[       OK ] Cdef/CDEFBlockTest.MatchTest/134 (128 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/135
[       OK ] Cdef/CDEFBlockTest.MatchTest/135 (83 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/136
[       OK ] Cdef/CDEFBlockTest.MatchTest/136 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/137
[       OK ] Cdef/CDEFBlockTest.MatchTest/137 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/138
[       OK ] Cdef/CDEFBlockTest.MatchTest/138 (83 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/139
[       OK ] Cdef/CDEFBlockTest.MatchTest/139 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/140
[       OK ] Cdef/CDEFBlockTest.MatchTest/140 (129 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/141
[       OK ] Cdef/CDEFBlockTest.MatchTest/141 (83 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/142
[       OK ] Cdef/CDEFBlockTest.MatchTest/142 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/143
[       OK ] Cdef/CDEFBlockTest.MatchTest/143 (130 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/144
[       OK ] Cdef/CDEFBlockTest.MatchTest/144 (160036 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/145
[       OK ] Cdef/CDEFBlockTest.MatchTest/145 (213793 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/146
[       OK ] Cdef/CDEFBlockTest.MatchTest/146 (251980 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/147
[       OK ] Cdef/CDEFBlockTest.MatchTest/147 (145 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/148
[       OK ] Cdef/CDEFBlockTest.MatchTest/148 (217 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/149
[       OK ] Cdef/CDEFBlockTest.MatchTest/149 (218 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/150
[       OK ] Cdef/CDEFBlockTest.MatchTest/150 (150 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/151
[       OK ] Cdef/CDEFBlockTest.MatchTest/151 (212 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/152
[       OK ] Cdef/CDEFBlockTest.MatchTest/152 (210 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/153
[       OK ] Cdef/CDEFBlockTest.MatchTest/153 (144 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/154
[       OK ] Cdef/CDEFBlockTest.MatchTest/154 (213 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/155
[       OK ] Cdef/CDEFBlockTest.MatchTest/155 (216 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/156
[       OK ] Cdef/CDEFBlockTest.MatchTest/156 (147 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/157
[       OK ] Cdef/CDEFBlockTest.MatchTest/157 (209 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/158
[       OK ] Cdef/CDEFBlockTest.MatchTest/158 (211 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/159
[       OK ] Cdef/CDEFBlockTest.MatchTest/159 (149 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/160
[       OK ] Cdef/CDEFBlockTest.MatchTest/160 (213 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/161
[       OK ] Cdef/CDEFBlockTest.MatchTest/161 (216 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/162
[       OK ] Cdef/CDEFBlockTest.MatchTest/162 (146 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/163
[       OK ] Cdef/CDEFBlockTest.MatchTest/163 (209 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/164
[       OK ] Cdef/CDEFBlockTest.MatchTest/164 (215 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/165
[       OK ] Cdef/CDEFBlockTest.MatchTest/165 (148 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/166
[       OK ] Cdef/CDEFBlockTest.MatchTest/166 (213 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/167
[       OK ] Cdef/CDEFBlockTest.MatchTest/167 (213 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/168
[       OK ] Cdef/CDEFBlockTest.MatchTest/168 (142 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/169
[       OK ] Cdef/CDEFBlockTest.MatchTest/169 (215 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/170
[       OK ] Cdef/CDEFBlockTest.MatchTest/170 (217 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/171
[       OK ] Cdef/CDEFBlockTest.MatchTest/171 (151 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/172
[       OK ] Cdef/CDEFBlockTest.MatchTest/172 (212 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/173
[       OK ] Cdef/CDEFBlockTest.MatchTest/173 (208 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/174
[       OK ] Cdef/CDEFBlockTest.MatchTest/174 (145 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/175
[       OK ] Cdef/CDEFBlockTest.MatchTest/175 (219 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/176
[       OK ] Cdef/CDEFBlockTest.MatchTest/176 (210 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/177
[       OK ] Cdef/CDEFBlockTest.MatchTest/177 (149 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/178
[       OK ] Cdef/CDEFBlockTest.MatchTest/178 (209 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/179
[       OK ] Cdef/CDEFBlockTest.MatchTest/179 (217 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/180
[       OK ] Cdef/CDEFBlockTest.MatchTest/180 (151 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/181
[       OK ] Cdef/CDEFBlockTest.MatchTest/181 (217 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/182
[       OK ] Cdef/CDEFBlockTest.MatchTest/182 (215 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/183
[       OK ] Cdef/CDEFBlockTest.MatchTest/183 (143 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/184
[       OK ] Cdef/CDEFBlockTest.MatchTest/184 (212 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/185
[       OK ] Cdef/CDEFBlockTest.MatchTest/185 (221 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/186
[       OK ] Cdef/CDEFBlockTest.MatchTest/186 (147 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/187
[       OK ] Cdef/CDEFBlockTest.MatchTest/187 (217 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/188
[       OK ] Cdef/CDEFBlockTest.MatchTest/188 (210 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/189
[       OK ] Cdef/CDEFBlockTest.MatchTest/189 (146 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/190
[       OK ] Cdef/CDEFBlockTest.MatchTest/190 (217 ms)
[ RUN      ] Cdef/CDEFBlockTest.MatchTest/191
[       OK ] Cdef/CDEFBlockTest.MatchTest/191 (222 ms)
[----------] 192 tests from Cdef/CDEFBlockTest (1541950 ms total)

[----------] 1 test from Cdef/CDEFFindDirTest
[ RUN      ] Cdef/CDEFFindDirTest.MatchTest/0
[       OK ] Cdef/CDEFFindDirTest.MatchTest/0 (20520 ms)
[----------] 1 test from Cdef/CDEFFindDirTest (20520 ms total)

[----------] Global test environment tear-down
[==========] 196 tests from 3 test cases ran. (1563481 ms total)
[  PASSED  ] 196 tests.
Program ended with exit code: 0
```

File: test/CdefTest.cc [hunk 1]
```diff
@@ -0,0 +1,558 @@
...
+ * second_strength: 0, 1, 2, 4
+ * primary_damping: [3, 6] + (bd_ - 8)
+ * second_damping: [3, 6] + (bd_ - 8)
+ * direction: [0, 7]
+ * bitdepth: 8, 10, 12
+ *
+ */
+class CDEFBlockTest : public ::testing::TestWithParam<cdef_dir_param_t> {
+  public:
+    CDEFBlockTest() : rnd_(0, (1 << 16) - 1) {
```
Comment (edmond-zhu) [inline test/CdefTest.cc:69]: why not write to rnd_(16, false)? it is easier to read.

File: test/CdefTest.cc [hunk 2]
```diff
@@ -0,0 +1,558 @@
...
+        }
+    }
+
+    void test_cdef(int iterations) {
+        int pri_damping, sec_damping, bits, level, count;
+        // for 8-bit content, damping ranges [3, 6] for luma, [2, 5] for chroma
+        // for highbd content, range will add the additinal bitdepth. 12-bit
+        // content will range [7, 10]
+        for (pri_damping = 3 + bd_ - 8;
+             pri_damping < 7 - 3 * !!boundary_ + bd_ - 8;
```
Comment (edmond-zhu) [inline test/CdefTest.cc]: can we move "7 - 3 * !!boundary_ + bd_ - 8" outside the loop and it seems constant during the processing in loop

Comment (Cidana-Developers) [reply test/CdefTest.cc]: Done.

File: test/CdefTest.cc (hunk 1 above)
Comment (Cidana-Developers) [reply test/CdefTest.cc:69]: The second parameter is confusing, hard to remember its signed or unsigned. I leave the range here to claim the max and min.

**Pull request merged**